examples/step-a/step-a.cc

```diff
@@ -0,0 +1,1654 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2000 - 2016 by the deal.II authors
+ *
+ * This file is part of the deal.II library.
+ *
+ * The deal.II library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the deal.II distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Wolfgang Bangerth, University of Heidelberg, 2000
+ *         Daniel Garcia-Sanchez, CNRS, 2018
+ */
+
+// @sect3{Include files}
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <hdf5.h>
+
+#include <fstream>
+#include <iostream>
+
+// @sect3{Hdf5 library}
+
+namespace hdf5
+{
+  class H5Group
```

Comment:
I haven't even tried to understand yet what this class does, but do you think that that might be generic functionality better suited for the library proper, rather than an individual tutorial program?

Reply:
This class provides hdf5 functionality. Create groups, datasets, save attributes, read attributes, etc.
I use it to store the flux vs the frequency across a section and then read the data with python.
I'll submit the class in a separate PR, but first I'll improve the library, add support for deal.ii Vectors/Tensors and add MPI support.

Reply:
Great, I think that would be useful if you could make it general enough to use in other contexts as well!